src/bindgen/ir/enumeration.rs

```diff
@@ -328,9 +328,7 @@ impl Enum {
             .repr
             .ty
             .map(|ty| ty.to_primitive())
-            // technically you're not going to get #[repr(double)] because that's not valid rust,
-            // but let's check anyway
-            .map_or(false, |prim| prim.can_be_enum_fixed_type());
+            .is_some();
```

Comment:
I believe this is not what you intended. This is effectively `self.repr.ty.is_some()` so it should be simplified to that probably.